pilot/pkg/model/network.go

```diff
@@ -108,40 +155,101 @@ func NewNetworkManager(env *Environment) *NetworkManager {
 		lcmVal = lcm(lcmVal, num)
 	}
 
-	return &NetworkManager{
-		lcm:                 uint32(lcmVal),
-		byNetwork:           byNetwork,
-		byNetworkAndCluster: byNetworkAndCluster,
+	mgr.lcm = uint32(lcmVal)
+	mgr.byNetwork = byNetwork
+	mgr.byNetworkAndCluster = byNetworkAndCluster
+
+	return gatewaySet
+}
+
+func (mgr *NetworkManager) resolveHostnameGateways(gatewaySet map[NetworkGateway]struct{}) {
+	// filter the list of gateways to resolve
+	hostnameGateways := map[string][]NetworkGateway{}
+	names := sets.NewSet()
+	for gw := range gatewaySet {
+		if gwIP := net.ParseIP(gw.Addr); gwIP != nil {
+			continue
+		}
+		delete(gatewaySet, gw)
+		if !features.HostnameGateways {
+			log.Warnf("Failed parsing gateway address %s from Service Registry. "+
+				"Hostnames are not supported for gateways",
```

Comment:
We should teach user how to enable this in the log if we don't feel confident to enable this support in day 1.

Reply:
Good point – although I'm thinking we can just turn the feature flag on by default.